src/OpenGL/Extensions/Silk.NET.OpenGL.Extensions.ImGui/Texture.cs

```diff
@@ -23,55 +23,66 @@ class Texture : IDisposable
-            _gl.TextureStorage2D(GlTexture, MipmapLevels, InternalFormat, Width, Height);
+            GlTexture = _gl.GenTexture();
+            Bind();
 
-            _gl.TextureSubImage2D(GlTexture, 0, 0, 0, Width, Height, PixelFormat.Bgra, PixelType.UnsignedByte, (void*) data);
+            _gl.TexStorage2D(GLEnum.Texture2D, MipmapLevels, InternalFormat, Width, Height);
+            _gl.TexSubImage2D(GLEnum.Texture2D, 0, 0, 0, Width, Height, PixelFormat.Bgra, PixelType.UnsignedByte, (void*) data);
 
             if (generateMipmaps) _gl.GenerateTextureMipmap(GlTexture);
 
             SetWrap(TextureCoordinate.S, TextureWrapMode.Repeat);
             SetWrap(TextureCoordinate.T, TextureWrapMode.Repeat);
 
-            _gl.TextureParameter(GlTexture, TextureParameterName.TextureMaxLevel, MipmapLevels - 1);
+            _gl.TexParameterI(GLEnum.Texture2D, TextureParameterName.TextureMaxLevel, MipmapLevels - 1);
+        }
+
+        public void Bind()
+        {
+            _gl.BindTexture(GLEnum.Texture2D, GlTexture);
         }
 
         public void SetMinFilter(TextureMinFilter filter)
         {
-            _gl.TextureParameter(GlTexture, TextureParameterName.TextureMinFilter, (int)filter);
+            Bind();
+            _gl.TexParameterI(GLEnum.Texture2D, TextureParameterName.TextureMinFilter, (int)filter);
         }
 
         public void SetMagFilter(TextureMagFilter filter)
         {
-            _gl.TextureParameter(GlTexture, TextureParameterName.TextureMagFilter, (int)filter);
+            Bind();
+            _gl.TexParameterI(GLEnum.Texture2D, TextureParameterName.TextureMagFilter, (int)filter);
         }
 
         public void SetAnisotropy(float level)
         {
+            Bind();
             const TextureParameterName textureMaxAnisotropy = (TextureParameterName)0x84FE;
-            _gl.TextureParameter(GlTexture, textureMaxAnisotropy, Util.Clamp(level, 1, MaxAniso.GetValueOrDefault()));
+            _gl.TexParameter(GLEnum.Texture2D, (GLEnum)textureMaxAnisotropy, Util.Clamp(level, 1, MaxAniso.GetValueOrDefault()));
         }
 
         public void SetLod(int @base, int min, int max)
         {
-            _gl.TextureParameter(GlTexture, TextureParameterName.TextureLodBias, @base);
-            _gl.TextureParameter(GlTexture, TextureParameterName.TextureMinLod, min);
-            _gl.TextureParameter(GlTexture, TextureParameterName.TextureMaxLod, max);
+            Bind();
+            _gl.TexParameterI(GLEnum.Texture2D, TextureParameterName.TextureLodBias, @base);
+            _gl.TexParameterI(GLEnum.Texture2D, TextureParameterName.TextureMinLod, min);
+            _gl.TexParameterI(GLEnum.Texture2D, TextureParameterName.TextureMaxLod, max);
         }
         
         public void SetWrap(TextureCoordinate coord, TextureWrapMode mode)
         {
-            _gl.TextureParameter(GlTexture, (TextureParameterName)coord, (int)mode);
+            Bind();
```

Comment:
This isn't _quite_ what I meant. Basically the problem is as follows. Let's say you do this:
```cs
texture.SetLod(...)
texture.SetWrap(...)
```

This evaluates to:
```
glBindTexture
glTexParameter
glBindTexture
glTexParameter
```

Which is a bit inefficient. The addition of the `Bind` method means that we can just chain these calls under one bind, so the usage would look like:
```cs
texture.Bind();
texture.SetLod(...)
texture.SetWrap(...)
```

The goal of this is to remove the implicit binding that these Set* functions had and move the texture binding to where we're actually using the texture functions, so that we have more control over when we bind the texture and, in cases where we're setting multiple texture parameters at a time we can eliminate the redundant bind calls.

Reply:
I totally understood what you were looking for, and yes this is inefficient.

However, I did it that way, because I thought the purpose of having a helper object in the first place was that it could be somewhat smart enough to handle its own binding.

Anyway, I'm cool with this, and I switched it back accordingly.